app/lib/backend.dart

```diff
@@ -289,11 +289,7 @@ class GCloudPackageRepository extends PackageRepository {
       package.updated = newVersion.created;
 
       // Keep the latest version in the package object up-to-date.
-      if (package.latestSemanticVersion < newVersion.semanticVersion &&
-          (package.latestSemanticVersion.isPreRelease ||
-              !newVersion.semanticVersion.isPreRelease)) {
-        package.latestVersionKey = newVersion.key;
-      }
+      package.updateVersion(newVersion.key);
```

Comment:
Since you have the `PackageVersion` available on new uploads, I think it would be better to pass it instead of just a `Key` object (see comment below)